packages/server/dataloader/azureDevOpsLoaders.ts

```diff
@@ -0,0 +1,425 @@
+  return new DataLoader<TeamUserKey, AzureUserInfo | undefined, string>(
+    async (keys) => {
+      const results = await Promise.allSettled(
+        keys.map(async ({userId, teamId}) => {
+          const auth = await parent
+            .get('teamMemberIntegrationAuths')
+            .load({service: 'azureDevOps', teamId, userId})
+
+          if (!auth) {
+            return undefined
+          }
+
+          const provider = await parent.get('integrationProviders').loadNonNull(auth.providerId)
+
+          const manager = new AzureDevOpsServerManager(
+            auth,
+            provider as IntegrationProviderAzureDevOps
+          )
+
+          const restResult = await manager.getMe()
+          const {error, azureDevOpsUser} = restResult
+          if (error !== undefined || azureDevOpsUser === undefined) {
+            console.log(error)
+            return undefined
+          }
+
+          return {
+            ...azureDevOpsUser
+          }
+        })
+      )
+      return results.map((result) => (result.status === 'fulfilled' ? result.value : undefined))
+    },
+    {
+      ...parent.dataLoaderOptions,
+      cacheKeyFn: (key) => `${key.teamId}:${key.userId}`
+    }
+  )
+}
+
+export const allAzureDevOpsAccessibleOrgs = (
+  parent: RootDataLoader
+): DataLoader<TeamUserKey, Resource[], string> => {
+  return new DataLoader<TeamUserKey, Resource[], string>(
+    async (keys) => {
+      const results = await Promise.allSettled(
+        keys.map(async ({userId, teamId}) => {
+          const auth = await parent
+            .get('teamMemberIntegrationAuths')
+            .load({service: 'azureDevOps', teamId, userId})
+
+          if (!auth) {
+            return []
+          }
+
+          const provider = await parent.get('integrationProviders').loadNonNull(auth.providerId)
+
+          const manager = new AzureDevOpsServerManager(
+            auth,
+            provider as IntegrationProviderAzureDevOps
+          )
+
+          const userInfo = await parent.get('azureDevUserInfo').load({teamId, userId})
+          if (!userInfo) return []
+          const {id} = userInfo
+
+          const results = await manager.getAccessibleOrgs(id)
+          const {error, accessibleOrgs} = results
+          // handle error if defined
+          console.log(error)
+          return accessibleOrgs.map((resource) => ({
+            ...resource
+          }))
+        })
+      )
+      return results.map((result) => (result.status === 'fulfilled' ? result.value : []))
+    },
+    {
+      ...parent.dataLoaderOptions,
+      cacheKeyFn: (key) => `${key.userId}:${key.teamId}`
+    }
+  )
+}
+
+export const allAzureDevOpsProjects = (
+  parent: RootDataLoader
+): DataLoader<TeamUserKey, TeamProjectReference[], string> => {
+  return new DataLoader<TeamUserKey, TeamProjectReference[], string>(
+    async (keys) => {
+      const results = await Promise.allSettled(
+        keys.map(async ({userId, teamId}) => {
```

Comment:
-1 Why not use `const auth = await parent.get('freshAzureDevOpsAuth').load({teamId, userId})`?

same for functions below

Reply:
changed